### PR TITLE
Resolve HLint 'Use Foo' hints

### PR DESCRIPTION
**Pull request by lwm**
Some questionable, some not. Shout and I will ignore them. 

Refer to [HLint manual](http://community.haskell.org/~ndm/darcs/hlint/hlint.htm) for configuration/ignore syntax.

Nearly there. Just another PR to go and then #2591 can move forward.

Ref: https://github.com/commercialhaskell/stack/issues/2592

### CODE REVIEW

File: src/Stack/Types/FlagName.hs [hunk 1]
```diff
@@ -54,7 +54,7 @@ newtype FlagName =
   FlagName Text
   deriving (Typeable,Data,Generic,Hashable,Store,NFData)
 instance Eq FlagName where
-    x == y = compare x y == EQ
+    x == y = x == y
```
Comment (sjakobi) [inline src/Stack/Types/FlagName.hs]: It's funny that hlint would suggest this. :)

I'd feel safer with the old version

File: test/integration/lib/StackTest.hs [hunk 2]
```diff
@@ -156,9 +148,7 @@ fileContentsMatch f1 f2 = do
     doesExist f2
     f1Contents <- readFile f1
     f2Contents <- readFile f2
-    if f1Contents == f2Contents
-          then return ()
-          else error
+    unless (f1Contents == f2Contents) $ error
                    ("contents do not match for " ++
                     show f1 ++
                     " " ++
```
Comment (sjakobi) [inline test/integration/lib/StackTest.hs]: The indentation should be reduced here.

File: src/Stack/Types/FlagName.hs (hunk 1 above)
Comment (mgsloan) [inline src/Stack/Types/FlagName.hs]: Yes, this will lead to an infinite loop

File: src/Stack/Docker.hs [hunk 3]
```diff
@@ -720,7 +719,7 @@ checkDockerVersion envOverride docker =
        _ -> throwM InvalidVersionOutputException
   where minimumDockerVersion = $(mkVersion "1.6.0")
         prohibitedDockerVersions = []
-        stripVersion v = fst $ break (== '-') $ dropWhileEnd (not . isDigit) v
+        stripVersion v = takeWhile (not . (== '-')) (dropWhileEnd (not . isDigit) v)
```
Comment (lwm) [inline src/Stack/Docker.hs]: Done!

File: test/integration/lib/StackTest.hs (hunk 2 above)
Comment (lwm) [inline test/integration/lib/StackTest.hs]: Nice one, fixed up.

**Comment by lwm** [comment]
Addressed the comments and squashed them. 

I can ignore `Use ++` hints if you rethink and feel strongly on that @sjakobi.

**Comment by mgsloan** [comment]
LGTM!  I added a commit to fix warnings + build on 7.8

**Comment by lwm** [comment]
Great! Thanks @mgsloan :rocket:

**Comment by sjakobi** [comment]
I think it would be better to merge PRs that don't pass CI into a different branch than master and fix them there. Developing based on a master branch that won't compile is no fun.

**Comment by lwm** [comment]
> Developing based on a master branch that won't compile is no fun.

Agreed!!!

**Comment by mgsloan** [comment]
True, but in this case it compiled, just didn't pass travis.  I am fine with having master states that don't pass travis, but I realize that this varies from some other people's workflow.

**Comment by lwm** [comment]
Oh, you don't use protected branches? Is that even less fun?

**Comment by mgsloan** [comment]
I just like to minimize overall development friction, and sometimes that means breaking travis occasionally.  We can certainly move to a more PR-centric development workflow if it becomes a problem.

**Pull request merged**